website/docs/faqs/Warehouse/db-connection-dbt-compile.md

```diff
@@ -0,0 +1,21 @@
+---
+title: Why does dbt compile need a data platform connection?
+description: "`dbt compile` needs a data platform connection because the work it does depends on the current state of your warehouse"
+sidebar_label: "Why does dbt compile need a data platform connection?"
+id: db-connection-dbt-compile
+---
+
+The [`dbt compile`](reference/commands/compile) command transforms raw SQL into dbt models and tests into compiled SQL. `compile` is similar to `run` except `compile` doesn't materialize the model's compiled SQL into an existing table. So, up until the point of materialization, `compile` and `run` are similar because they both require a data platform connection, run queries, and have an [`execute` variable](/reference/dbt-jinja-functions/execute) set to `True`. A few things to note:
```

Comment:
```suggestion
The [`dbt compile`](reference/commands/compile) command generates executable SWL from source, `model`, `test`, and `analysis` files. 

`dbt compile` is similar to `dbt run` except that it doesn't materialize the model's compiled SQL into an existing table. So, up until the point of materialization, `dbt compile` and `dbt run` are similar because they both require a data platform connection, run queries, and have an [`execute` variable](/reference/dbt-jinja-functions/execute) set to `True`. There are a few things to note:
```